modules/data-entry/src/main/java/ca/sickkids/ccm/lfs/PaginationServlet.java

```diff
@@ -72,7 +72,14 @@ public void doGet(final SlingHttpServletRequest request, final SlingHttpServletR
         final String filter = request.getParameter("filter");
         if (StringUtils.isNotBlank(filter)) {
             // Full text search; \ and ' must be escaped
-            query.append(" and contains(*, '" + filter.replaceAll("['\\\\]", "\\\\$0") + "')");
+            query.append(" and contains(n.*, '" + filter.replaceAll("['\\\\]", "\\\\$0") + "')");
+        }
+        final String fieldname = request.getParameter("fieldname");
+        final String fieldvalue = request.getParameter("fieldvalue");
+        if (StringUtils.isNotBlank(fieldname) && StringUtils.isNotBlank(fieldvalue)) {
+            // Exact condition; \ and ' must be escaped. The value must be wrapped in 's
+            query.append(" and " + fieldname.replaceAll("['\\\\]", "\\\\$0") + "='"
```

Comment:
We escape `'` and `\` for the value so that it's not possible to get out of the quotes. But the field name is not quoted, so this escaping doesn't do anything useful. The good news is that the query syntax allows something like `where n.'weird property' = 'weird value'`.

```suggestion
            query.append(" and n.'" + fieldname.replaceAll("['\\\\]", "\\\\$0") + "'='"
```